rust/lance-core/src/datatypes/schema.rs

```diff
@@ -113,7 +113,19 @@ impl Schema {
                 ), location: location!(),});
             }
         }
-        Ok(true)
+
+        // Check for duplicate field ids
+        let mut seen_ids = HashSet::new();
+        for field in self.fields_pre_order() {
+            if !seen_ids.insert(field.id) {
+                return Err(Error::Schema {
+                    message: format!("Duplicate field id {} in schema {:?}", field.id, self),
+                    location: location!(),
+                });
+            }
+        }
```

Comment:
How would this happen?  Or is this just a sanity check?

Reply:
Yeah this is just a sanity check. I was going to add a method `Schema::union(&self, other: &Schema)`, and needed to check we don't get duplicate field ids there. But I didn't end up needing that method. Still seems like something we should validate when we can.